task/host-sp-comms/src/bsp/gimlet_bcde.rs

```diff
@@ -405,23 +404,21 @@ impl ServerImpl {
             }
 
             52..=57 => {
-                let (connector_name, f, sensors): ([u8; 4], _, _) =
-                    match index - 52 {
-                        0 => by_refdes!(J194, tmp117),
-                        1 => by_refdes!(J195, tmp117),
-                        2 => by_refdes!(J196, tmp117),
-                        3 => by_refdes!(J197, tmp117),
-                        4 => by_refdes!(J198, tmp117),
-                        5 => by_refdes!(J199, tmp117),
-                        _ => unreachable!(),
-                    };
+                let (mut name, f, sensors): ([u8; 7], _, _) = match index - 52 {
+                    0 => by_refdes!(J194_U1, tmp117),
+                    1 => by_refdes!(J195_U1, tmp117),
+                    2 => by_refdes!(J196_U1, tmp117),
+                    3 => by_refdes!(J197_U1, tmp117),
+                    4 => by_refdes!(J198_U1, tmp117),
+                    5 => by_refdes!(J199_U1, tmp117),
+                    _ => unreachable!(),
+                };
                 let dev = f(I2C.get_task_id());
 
-                // Convert the name from Jxxx (in the TOML file) -> Jxxx/U1
-                let mut name = *b"Jxxx/U1";
-                // All connector names should have length 4; that's checked by
-                // the type in the tuple assignment above.
-                name[..4].copy_from_slice(&connector_name);
+                // Convert the name from Jxxx_U1 (in the TOML file) -> Jxxx/U1
+                // TODO(eliza): let's just get this from the include_refdes i2c
+                // codegen...
```

Comment:
It would be nice to change most of this code in `host-sp-comms` to just use the component ID strings we generate (added in #2189). However, I felt like this was better done in a follow-up PR as I think it requires some additional codegen changes (I'd like to make the component ID not be optional, which breaks a lot of places where we construct I2C devices "manually").